File: Accelrometerlearn/app/src/main/java/com/example/accelrometerlearn/MainActivity.java
package com.example.accelrometerlearn;

import androidx.appcompat.app.AppCompatActivity;

import android.hardware.Sensor;
import android.hardware.SensorEvent;
import android.hardware.SensorEventListener;
import android.hardware.SensorManager;
import android.os.Bundle;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity implements SensorEventListener {
TextView txt;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        SensorManager sensorManager=(SensorManager) getSystemService(SENSOR_SERVICE);
        if(sensorManager!=null){
            Sensor accel = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
            if(accel!=null){
                sensorManager.registerListener(this,accel,SensorManager.SENSOR_DELAY_NORMAL);
            }
        }else{
            Toast.makeText(this,"Sensor Services not detected",Toast.LENGTH_LONG).show();
        }
    }

    @Override
    public void onSensorChanged(SensorEvent sensorEvent) {
if(sensorEvent.sensor.getType()==Sensor.TYPE_ACCELEROMETER){
    txt=findViewById(R.id.txt);
    txt.setText("X: "+sensorEvent.values[0]+" Y: "+sensorEvent.values[1]+" Z: "+sensorEvent.values[2]);
}
    }

    @Override
    public void onAccuracyChanged(Sensor sensor, int i) {

    }
}

File: CustomAlertBox/app/src/main/java/com/example/customalertbox/MainActivity.kt
package com.example.customalertbox


import android.app.Dialog


import android.os.Bundle
import android.widget.Button
import android.widget.Toast
import androidx.activity.ComponentActivity

import com.example.customalertbox.R.layout.custom_dialog
import com.example.customalertbox.databinding.ActivityMainBinding

class MainActivity : ComponentActivity() {
    private lateinit var binding:ActivityMainBinding
    private lateinit var dialog: Dialog
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding=ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        dialog=Dialog(this)
        dialog.setContentView(custom_dialog)
        dialog.window?.setBackgroundDrawable(getDrawable(R.drawable.alertbox))


        val goodButton =dialog.findViewById<Button>(R.id.goodbut)
            val badButton =dialog.findViewById<Button>(R.id.badbut)

        goodButton.setOnClickListener {
            dialog.dismiss()
        }

        badButton.setOnClickListener {

          Toast.makeText(this,"Try Kar Mil Samjhega",Toast.LENGTH_LONG).show()
        }

        binding.but1.setOnClickListener {

            dialog.show()
        }




    }


}

